core/schema/sdk.go

```diff
@@ -633,6 +633,34 @@ func (sdk *goSDK) baseWithCodegen(
 				},
 			},
 		},
+	}
+
+	// inject sdk specific env variables before withExec
+	// for now allow only Env with prefix GO
+	cfg, ok, _ := src.Self.ModuleConfig(ctx)
+	if ok && cfg.SDK != nil {
+		for k, v := range cfg.SDK.Env {
+			if !strings.HasPrefix(k, "GO") {
```

Comment:
Do we want to add this condition?  I think this will be confusing to some users. I don't see why we would like this to be this way.  For example, there are some Go related variables that don't start with `GO` like `CGO_ENABLED=1`.

Reply:
that is a fair point. are you running into a limitation already because of that? The idea was we can extend it on case by case basis to ensure we don't end up allowing something and then having to revert that later (e.g. for security considerations).

Reply:
> that is a fair point. are you running into a limitation already because of that? The idea was we can extend it on case by case basis to ensure we don't end up allowing something and then having to revert that later (e.g. for security considerations).

oh, I see what you mean.. yeah.. I don't have any particular requirement now. Given the security constraints, it seems ok to me to leave it like this for now :pray:

Reply:
Split this to https://github.com/dagger/dagger/pull/9558.